packages/ui-components/src/components/absolute-time-picker/absolute-time-picker.tsx

```diff
@@ -0,0 +1,405 @@
+import dayjs from 'dayjs';
+import { CALENDAR_MASK, DATETIME_INPUT_MASK, DATE_INPUT_MASK, DATE_INPUT_PLACEHOLDER, DEFAULT_HEADER_TITLE } from './absolute-time-picker.config';
+import { formatDateTime, fromDateInput, fromISO, isDateAfter, isDateBefore, isDateSame, isDateTimeAfter, newDate } from '../../utils/date.helper';
+import { isEmpty } from 'lodash';
+import { ERelativeTimeInputMode, IAbsoluteTimePicker, IAbsoluteTimePickerEvents, IRelativeTimeInput, IAbsoluteSelectedRangeDates } from './absolute-time-picker.types';
+import { buildSelectedDatesEventPayload } from './absolute-time-picker.helper';
+
+@Component({
+	tag: 'kv-absolute-time-picker',
+	styleUrl: 'absolute-time-picker.scss',
+	shadow: true
+})
+export class KvAbsoluteTimePicker implements IAbsoluteTimePicker, IAbsoluteTimePickerEvents {
+	/** @inheritdoc */
+	@Prop({ reflect: false }) headerTitle?: string = DEFAULT_HEADER_TITLE;
+	/** @inheritdoc */
+	@Prop({ reflect: false }) displayBackButton?: boolean = false;
+	/** @inheritdoc */
+	@Prop({ reflect: false }) selectedRangeDates?: string[] = [];
+	/** @inheritdoc */
+	@Prop({ reflect: false }) initialDate?: string;
+	/** @inheritdoc */
+	@Prop({ reflect: false }) disabledDates?: string[] = [];
+	/** @inheritdoc */
+	@Prop({ reflect: false }) relativeTimeConfig?: IRelativeTimeInput;
+
+	/** @inheritdoc */
+	@Event() selectRangeDatesChange: EventEmitter<IAbsoluteSelectedRangeDates>;
+	/** @inheritdoc */
+	@Event() backButtonClicked: EventEmitter<MouseEvent>;
+	/** @inheritdoc */
+	@Event() relativeTimeConfigReset: EventEmitter<MouseEvent>;
+	/** @inheritdoc */
+	@Event() relativeTimeConfigChange: EventEmitter<IAbsoluteSelectedRangeDates>;
+
+	@State() fromInputValue: string = '';
+	@State() toInputValue: string = '';
+	@State() hoveredDate: string = '';
+	@State() displayedMonth: dayjs.Dayjs = fromDateInput(this.initialDate) ?? fromDateInput(new Date());
+	@State() fromInputFocused: boolean = true;
+	@State() toInputFocused: boolean = false;
+	@State() inputMode: ERelativeTimeInputMode = ERelativeTimeInputMode.Date;
+	@State() maxDate: string = '';
+	@State() minDate: string = '';
+
+	@Watch('relativeTimeConfig')
+	handleRelativeTimeConfigInput(newValue: IRelativeTimeInput) {
+		if (isEmpty(newValue)) {
+			this.inputMode = ERelativeTimeInputMode.Date;
+			if (this.selectedRangeDates.length === 0) {
+				this.fromInputValue = '';
+				this.toInputValue = '';
+				this.displayedMonth = fromDateInput(new Date());
+			}
+			return;
+		}
+
+		this.inputMode = newValue.mode;
+		this.fromInputValue = newValue.from;
+		this.toInputValue = newValue.to;
+		const [from, _to] = this.selectedRangeDates;
+		const date = newDate(from);
+		if (date.isValid()) {
+			this.displayedMonth = date;
+		}
+	}
+
+	private handleBackClick = (event: MouseEvent) => {
+		this.backButtonClicked.emit(event);
+	};
+
+	public onStartDateChange = (event: CustomEvent<string>) => {
+		const date = event.detail;
+		const parsedDate = dayjs(date, DATE_INPUT_MASK);
```

Comment:
Because we must handle two different dates! When the date does not have time, we should add by default the 00:00:00 or the 23:59:59. Therefore, this validation demands two steps.